mudz.Common/Domain/Player/PlayerEqualityComparer.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace mudz.Common.Domain.Player
+{
+	public sealed class PlayerEqualityComparer : Singleton<PlayerEqualityComparer>, IEqualityComparer<IPlayer>
```

Comment:
Simple Equality comparer to perform equality against the name instead of reference. Singleton'd and can't be new'd up due to the inheritance, so minimal memory footprint.